src/test/libcephfs/test.cc

```diff
@@ -2428,3 +2428,19 @@ TEST(LibCephFS, Lseek) {
   ASSERT_EQ(0, ceph_close(cmount, fd));
   ceph_shutdown(cmount);
 }
+
+TEST(LibCephFS, LookupInoMDSDir) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, NULL), 0);
+
+  Inode *inode;
+  auto ino = inodeno_t(0x100); /* rank 0 ~mdsdir */
```

Comment:
This causes the following error in Yuri's builds https://shaman.ceph.com/builds/ceph/wip-yuri8-testing-2021-04-13-1123-nautilus/,
```
[ 82%] Linking CXX executable ../../../bin/ceph_test_libcephfs_lazyio
Scanning dependencies of target rbd
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/test/libcephfs/test.cc: In member function 'virtual void LibCephFS_LookupInoMDSDir_Test::TestBody()':
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/test/libcephfs/test.cc:2440:29: error: invalid use of incomplete type 'struct inodeno_t'
   auto ino = inodeno_t(0x100); /* rank 0 ~mdsdir */
                             ^
In file included from /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/include/cephfs/libcephfs.h:30,
                 from /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/test/libcephfs/test.cc:17:
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/include/cephfs/ceph_ll_client.h:23:8: note: forward declaration of 'struct inodeno_t'
 struct inodeno_t;
        ^~~~~~~~~
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/test/libcephfs/test.cc:2442:24: error: invalid use of incomplete type 'struct inodeno_t'
   ino = inodeno_t(0x600); /* rank 0 first stray dir */
                        ^
In file included from /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/include/cephfs/libcephfs.h:30,
                 from /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/test/libcephfs/test.cc:17:
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/14.2.19-123-g26631556/rpm/el8/BUILD/ceph-14.2.19-123-g26631556/src/include/cephfs/ceph_ll_client.h:23:8: note: forward declaration of 'struct inodeno_t'
 struct inodeno_t;
        ^~~~~~~~~
```  
https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/53756//consoleFull

Reply:
@batrick 
I think this is failing due to missing header file,
https://github.com/ceph/ceph/pull/39294/commits/a956c7986fd43da851918b87c71524f9a2bb1162#diff-9751a84845e9700a36244d126245bd8233894a8e0fdf0ef340545b894ea2f2e4R18

The reference commit is part of PR https://github.com/ceph/ceph/pull/39294 that needs to be backported to nautilus. The backport is not trivial. We might need the PR's original author to backport it, before backporting this PR.